migtools/oadp-non-admin
# fix: add integration tests for NAB

## Why the changes were made

Add Integration tests for NAB, so we can trust our code base more

Fixes https://github.com/migtools/oadp-non-admin/issues/61

Also
- removed duplication in logs
- removed unnecessary validations
- made each reconcile only execute one update call per object, to avoid race conditions
- removed terminal error from wrong places 

Can help on refactor of https://github.com/migtools/oadp-non-admin/issues/62

## How to test the changes made

To run integration tests (and also unit tests), run `make simulation-test` 

Check Test logs to see what is being tested

To check code coverage, run `go tool cover -html=cover.out`

Also check https://github.com/migtools/oadp-non-admin/pull/73#discussion_r1747386167
